src/transformers/models/gpt2/modeling_flax_gpt2.py

```diff
@@ -527,12 +524,23 @@ def __call__(
         hidden_states = outputs[0]
         hidden_states = self.ln_f(hidden_states)
 
+        all_hidden_states = None
+        if output_hidden_states:
+            if not return_dict:
+                all_hidden_states = outputs[1]
+            else:
+                all_hidden_states = outputs.hidden_states
+            all_hidden_states = all_hidden_states + (hidden_states,)
+
```

Comment:
```suggestion
        if output_hidden_states:
            all_hidden_states = outputs[1] + (hidden_states,)
            outputs = (hidden_states, all_hidden_states) + outputs[2:]
        else:
            outputs = (hidden_states,) + outputs[1:]
```

The model output classes support indexing, so we don't need to check `return_dict` here